src/types/createResolve.ts

```diff
@@ -2,13 +2,13 @@ import { GraphQLFieldConfig, GraphQLFieldResolver } from 'graphql';
 
 import NodeType from './NodeType';
 
-export default function createResolve<TFields extends string, TContext = any>(
-  resolve: GraphQLFieldResolver<Record<TFields, any>, TContext>,
-  fieldNames: TFields[],
-): GraphQLFieldConfig<any, TContext>['resolve'] {
+export default function createResolve<TSource, TContext = any>(
+  resolve: GraphQLFieldResolver<TSource, TContext>,
+  ensureFields: (keyof TSource)[],
+): GraphQLFieldConfig<TSource, TContext>['resolve'] {
```

Comment:
One more thing -- won't this make the resolve's object argument be typed with all the source fields? 
Ideally we want the object to be `keys of ensureFields` and `id`

Reply:
the object that is passed in isn't jsut the fields you requested it's always the full node. The fields array is a maybe-to-complex hint to the helper on whether it needs to fetch the node or not. e.g. if `favoriteColor` is already on the source node don't make a request